src/pybind/mgr/dashboard/controllers/rgw.py

```diff
@@ -742,6 +758,35 @@ def get_lifecycle_policy(self, bucket_name: str = '', daemon_name=None, owner=No
         bucket_name = RgwBucket.get_s3_bucket_name(bucket_name, tenant)
         return self._get_lifecycle(bucket_name, daemon_name, owner)
 
+    @RESTController.Collection(method='GET', path='/notification')
+    @allow_empty_body
+    @EndpointDoc("Get the bucket notification")
+    def get_notification(self, bucket_name: str = '', notification_id: Optional[str] = None,
+                         daemon_name=None, owner=None):
+        owner = _get_owner(owner)
+        bucket_name = RgwBucket.get_s3_bucket_name(bucket_name)
+        return self._get_notification(bucket_name, notification_id, daemon_name, owner)
+
+    @RESTController.Collection(method='PUT', path='/notification')
+    @allow_empty_body
+    @EndpointDoc("Create or update the bucket notification")
+    def set_notification(self, bucket_name: str = '', notification: str = '', daemon_name=None,
+                         owner=None):
+        owner = _get_owner(owner)
+        bucket_name = RgwBucket.get_s3_bucket_name(bucket_name)
+        if notification == '{}':
+            return self._delete_notification(bucket_name, daemon_name, owner)
+        return self._set_notification(bucket_name, notification, daemon_name, owner)
+
+    @RESTController.Collection(method='DELETE', path='/notification')
+    @allow_empty_body
```

Comment:
I don 't think `@allow_empty_body` is needed
```suggestion
    @EndpointDoc("Get the bucket notification")
    def get_notification(self, bucket_name: str = '', notification_id: Optional[str] = None,
                         daemon_name=None, owner=None):
        owner = _get_owner(owner)
        bucket_name = RgwBucket.get_s3_bucket_name(bucket_name)
        return self._get_notification(bucket_name, notification_id, daemon_name, owner)

    @RESTController.Collection(method='PUT', path='/notification')
    @EndpointDoc("Create or update the bucket notification")
    def set_notification(self, bucket_name: str = '', notification: str = '', daemon_name=None,
                         owner=None):
        owner = _get_owner(owner)
        bucket_name = RgwBucket.get_s3_bucket_name(bucket_name)
        if notification == '{}':
            return self._delete_notification(bucket_name, daemon_name, owner)
        return self._set_notification(bucket_name, notification, daemon_name, owner)

    @RESTController.Collection(method='DELETE', path='/notification')
```

Reply:
![image](https://github.com/user-attachments/assets/daa38f9e-5888-4944-af10-3d05fa9dc173)

What is the purpose of the @allow_empty_body decorator in the context of lifecycle methods, and why might it be applied there? I have taken reference from the same method

Reply:
Suggested changes are made